graphqlapi/src/main/java/io/stargate/graphql/schema/fetchers/ddl/KeyspaceFormatter.java

```diff
@@ -69,13 +96,30 @@
       Supplier<Iterable<ChildT>> allChildrenGetter,
       Function<String, ChildT> childByNameGetter,
       Function<ChildT, Map<String, Object>> converter,
-      DataFetchingEnvironment environment) {
+      DataFetchingEnvironment environment,
+      String keyspaceName,
+      AuthorizationService authorizationService,
+      String token) {
 
     // All children query, for example `keyspace(name: "ks") { tables }`
     String allChildrenName = childFieldName + "s";
     if (environment.getSelectionSet().getField(allChildrenName) != null) {
       List<Map<String, Object>> formattedChildren = new ArrayList<>();
       for (ChildT child : allChildrenGetter.get()) {
+        if (childFieldName.equals("table")) {
+          try {
+            authorizationService.authorizeSchemaRead(
+                token,
+                Collections.singletonList(keyspaceName),
+                Collections.singletonList(((ImmutableTable) child).name()));
+          } catch (UnauthorizedException e) {
+            LOG.info(
+                "Not returning table {}.{} due to not being authorized",
```

Comment:
Nit: this method is used both for tables and UDTs, so
```suggestion
                "Not returning table/UDT {}.{} due to not being authorized",
```
Although this is just a log, so maybe it doesn't matter.

Reply:
I'm only checking tables right now with the `if(childFieldName.equals("table"))` up above